tests/services/kernels/test_api.py

```diff
@@ -240,3 +250,28 @@
 async def test_config2(serverapp):
     assert serverapp.kernel_manager.allowed_message_types == []
 
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 6),
+    reason="Kernel manager is AsyncMappingKernelManager, Python version < 3.6"
+)
+async def test_async_kernel_manager(configurable_serverapp):
+    argv = ['--ServerApp.kernel_manager_class=jupyter_server.services.kernels.kernelmanager.AsyncMappingKernelManager']
+    app = configurable_serverapp(argv=argv)
+    assert isinstance(app.kernel_manager, AsyncMappingKernelManager)
+
+
+@pytest.fixture()
+def no_async_kernel_manager(monkeypatch):
+    """Remove jupyter_client.multikernelmanager.AsyncMultiKernelManager."""
+    monkeypatch.delattr("jupyter_client.multikernelmanager.AsyncMultiKernelManager")
+    monkeypatch.delattr("jupyter_server.services.kernels.kernelmanager.AsyncMappingKernelManager")
+
+@pytest.mark.skipif(
+    sys.version_info < (3, 6),
+    reason="Kernel manager is AsyncMappingKernelManager, Python version < 3.6"
+)
+async def test_async_kernel_manager_not_available(configurable_serverapp, no_async_kernel_manager):
+    argv = ['--ServerApp.kernel_manager_class=jupyter_server.services.kernels.kernelmanager.AsyncMappingKernelManager']
+    with pytest.raises(SystemExit):
+        app = configurable_serverapp(argv=argv)
```

Comment:
I think this test should be opened up to 3.5.  Then conditionally assert that it raises `ValueError` on 3.5.  Or add a test for 3.5 stuff, but then it must be skipped for > 3.5, so we're kinda in the same boat.  This would mean that on 3.5, this test is really testing the 3.5 restriction, while on other versions, it's testing an older jupyter_client being present.  Seems like a good use of "resources". :smile:

Given that the `no_async_kernel_manager` fixture will now be removing the `AsyncMultiKernelManager` only, we should make sure this scenario is doing the right thing.  I suppose the crash cause will move to the import of `AsyncMultiKernelManager` failing in kernelmanager.py.  Since we require jupyter_client >= 6.1, I don't think it's urgent that we make this more graceful.

Reply:
Done.